internal/clients/http/ec2/ec2_client.go

```diff
@@ -172,6 +176,45 @@ func (c *ec2Client) ImportPubkey(ctx context.Context, key *models.Pubkey, tag st
 	return *output.KeyPairId, nil
 }
 
+func (c *ec2Client) updatePubkeyTag(ctx context.Context, key *models.Pubkey, tag string) (string, error) {
+	ctx, span := otel.Tracer(TraceName).Start(ctx, "updatePubkeyTag")
+	defer span.End()
+
+	if !c.assumed {
+		return "", http.ServiceAccountUnsupportedOperationErr
+	}
+	logger := logger(ctx)
+	logger.Trace().Msgf("Fetching AWS key with fingerprint '%s' to update its tag", key.Fingerprint)
+	input := &ec2.DescribeKeyPairsInput{}
+	input.Filters = []types.Filter{{Name: ptr.To("fingerprint"), Values: []string{key.Fingerprint}}}
+	output, err := c.ec2.DescribeKeyPairs(ctx, input)
+	if err != nil {
+		if isAWSUnauthorizedError(err) {
+			err = clients.UnauthorizedErr
+		}
+		span.SetStatus(codes.Error, err.Error())
+		return "", fmt.Errorf("cannot fetch SSH key to update its tag %s: %w", key.Fingerprint, err)
+	}
+
+	if len(output.KeyPairs) == 0 {
+		return "", fmt.Errorf("SSH key not found by its fingerprint (%s): %w", key.Fingerprint, http.PubkeyNotFoundErr)
+	}
+	tagInput := &ec2.CreateTagsInput{
+		Resources: []string{*output.KeyPairs[0].KeyPairId},
```

Comment:
I think it will be more readable to save *output.KeyPairs[0] as a variable

Reply:
Good suggestion, but I've refactored the code completely, so now it doesn't make sense as it's used once ^_^